lib/service.middleware/apiviewRoutes.js

```diff
@@ -273,6 +273,11 @@ class ApiViewRoutes extends ServiceMiddleware {
 
   // TODO: to many input, need some work
   _handlerWrapper (handlerFunc, resolved, skipOnError, service, controller, orgOutput) {
+    // do nothing if a response has already been sent
+    if (resolved['$rawResponse'].headersSent) {
+      return when.resolve({});
+    }
```

Comment:
This doesn't seam to be needed, if the header write is handled in framework ("http.framework.express").
It will skip the rest of the chain once a header has been written.

Reply:
Just kidding, something like that does exist: [`res.finished`](https://nodejs.org/api/http.html#http_response_finished).

I missed this earlier because it's not included included in the Express documentation; it's part of the underlying Node response object.  I think replacing the `res.headersSent` in this PR with `res.finished` would address both of our concerns.